go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -174,11 +182,14 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 
 			// Forcibly demote the old master in topology, since we can't rely on the
 			// old master to be up to change its own record.
+			// Call UpdateTabletFields instead of ChangeType so that we can check the type
+			// before changing it and avoid unnecessary topo updates
```

Comment:
We can make ChangeType also avoid unnecessary topo updates. I think it would be good to consolidate even these into ChangeType.

Note that we'll need to be careful that we only return `NoUpdateNeeded` if both the type and timestamp match what we want. Also, we'll need to capture the tablet value so we can return it if we succeed due to NoUpdateNeeded, because otherwise I think UpdateTabletFields will return `nil`.

Reply:
If NoUpdateNeeded, then we don't need to refresh the tablet, so it should be ok to get back a nil tablet?

Reply:
Done.